core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -7971,15 +7971,17 @@ private void checkLiteral2(String expression, String expected) {
         + "\"birth_date\" = cast('1914-02-02' as date) or "
         + "\"hire_date\" = cast('1996-01-01 '||'00:00:00' as timestamp) or "
         + "\"hire_date\" = '1996-01-01 00:00:00' or "
-        + "cast(\"full_name\" as timestamp) = \"hire_date\"";
+        + "cast(\"full_name\" as timestamp) = \"hire_date\" or "
+        + "cast('10' as varchar) = 1";
```

Comment:
Now I get it. The result would be `cast('10' as varchar) = 1` without this changing, and It throws exception in PostgreSQL.